lib/Driver.js

```diff
@@ -46,34 +48,28 @@ module.exports = class Driver {
     }
 
     const [didAuthority, keyIdFragment] = did.split('#');
-
-    let addedToCache = false;
-    let promise = this._cache.get(didAuthority);
-    if(!promise) {
-      const fingerprint = didAuthority.substr('did:key:'.length);
-      const publicKey = LDKeyPair.fromFingerprint({fingerprint});
-
-      promise = this.keyToDidDoc(publicKey);
-      this._cache.set(didAuthority, promise);
-      addedToCache = true;
-    }
-
-    let didDoc;
     try {
-      didDoc = await promise;
-    } catch(e) {
-      if(addedToCache) {
-        this._cache.del(didAuthority);
+      // cache the result of expansion
+      const {didDocument} = await this._cache.memoize({
+        key: didAuthority,
+        fn: () => {
+          const fingerprint = didAuthority.substr('did:key:'.length);
+          const keyPair = LDKeyPair.fromFingerprint({fingerprint});
+          return this.keyPairToDidDocument({keyPair});
+        }
+      });
+
+      if(keyIdFragment) {
+        // Resolve an individual key
+        return _getKey({didDocument, keyIdFragment});
       }
-      throw e;
-    }
 
-    if(keyIdFragment) {
-      // Resolve an individual key
-      return getKey({didDoc, keyIdFragment});
+      // Resolve the full DID Document
+      return didDocument;
+    } catch(e) {
+      this._cache.delete(didAuthority);
+      throw e;
```

Comment:
Delete this, shouldn't need to mess with the cache, lru-memoize handles it.

Reply:
Deleted.